src/WebForms/Compilation/BaseTemplateParser.cs

```diff
@@ -66,7 +66,9 @@ protected internal Type GetUserControlType(string virtualPath)
 
     internal Type GetUserControlType(VirtualPath virtualPath)
     {
-        Type t = GetReferencedType(virtualPath, false /*allowNoCompile*/);
+        var t = GetReferencedType(virtualPath, false /*allowNoCompile*/);
+
+        t ??= CompiledTypeAccessor.GetForPath(virtualPath.Path);
```

Comment:
Wouldn't it make sense for the CompiledTypAccessor to be used within the GetReferencedType method?

Reply:
Yup, but things like this are where we need to swap out existing logic for how we want to integrate it into the new stack. I haven't done a great job with documenting that (should go back and do so), but feel free to add a comment such as `// This was done differently in the original WebForms, we changed it for x,y,z` or something like that

Reply:
@twsouthwick we can't.. it's coming in the path of MasterPage... for masterpage if return type is null we add TypedReference and that logic is failing when I move this code to GetReferencedType method... we may need to revert Master page logic old way.. feel free to create an issue if you think we should handle master page in original way.

Reply:
I've created an issue to track this